crates/bitcoin-da/src/verifier.rs

```diff
@@ -221,64 +222,167 @@ impl DaVerifier for BitcoinVerifier {
+            block_header,
+            latest_da_state,
+            target,
+            latest_da_state.current_target_bits,
+        )?;
 
-        // Check 2: block heights are consecutive
-        if block_header.height() - 1 != previous_light_client_proof_output.da_block_height {
-            return Err(ValidationError::NonConsecutiveBlockHeight);
-        }
-        // Check 3: prev hash matches with prev light client proof hash
-        if block_header.prev_hash() != previous_light_client_proof_output.da_block_hash {
-            return Err(ValidationError::InvalidPrevBlockHash);
+        let epoch_block = block_header.height() % BLOCKS_PER_EPOCH;
+        // Check if this is epoch block, and update time accordingly
+        let mut epoch_start_time = latest_da_state.epoch_start_time;
+        if epoch_block == 0 {
+            epoch_start_time = block_header.time().secs() as u32;
         }
-        // Check 4: valid bits
-        if block_header.bits() != previous_light_client_proof_output.da_current_target_bits {
-            return Err(ValidationError::InvalidBlockBits);
+
+        // Update previous timestamps
+        let mut prev_11_timestamps = latest_da_state.prev_11_timestamps;
+        prev_11_timestamps[block_header.height() as usize % 11] = block_header.time().secs() as u32;
+
+        // If the next block is epoch start block, calculate the next epoch's difficulty target
+        let mut current_target_bits = block_header.bits();
+        if epoch_block == BLOCKS_PER_EPOCH - 1 {
+            let next_target = calculate_new_difficulty(
+                epoch_start_time,
+                block_header.time().secs() as u32,
+                block_header.bits(),
+                network_constants.max_target,
+            );
+            current_target_bits = target_to_bits(&next_target);
         }
-        // Check 5: proof of work
-        if !verify_target_hash(block_header.hash().into(), target) {
-            return Err(ValidationError::InvalidTargetHash);
+
+        let total_work = U256::from_be_bytes(latest_da_state.total_work)
+            .saturating_add(&work_add)
+            .to_be_bytes();
+
+        Ok(LatestDaState {
+            block_hash: block_header.hash().to_byte_array(),
+            block_height: block_header.height(),
+            total_work,
+            epoch_start_time,
+            prev_11_timestamps,
+            current_target_bits,
+        })
+    }
+
+    fn verify_header_chain_testnet4(
+        &self,
+        latest_da_state: Option<&LatestDaState>,
+        block_header: &HeaderWrapper,
+    ) -> Result<LatestDaState, ValidationError> {
+        let network_constants = TESTNET4_CONSTANTS;
+        let latest_da_state = latest_da_state.unwrap_or(&INITIAL_TESTNET4_STATE);
+
+        let latest_block_time =
+            latest_da_state.prev_11_timestamps[latest_da_state.block_height as usize % 11];
+        // If more than 20 minutes passed since latest block, reset target
```

Comment:
This does not have to be true for all cases. It is true that the difficulty drops to minimum, but a real block with longer time larger than 20 minutes will still pass. See [block 40320](https://mempool.space/testnet4/block/0000000000000011e7cf114aa4980cbd74c2c07fccc8b025480dd4b7e1d96110)

Reply:
added epoch block check